various-receipes/data-isolation/anti-pattern-data-isolation.test.js

```diff
@@ -0,0 +1,100 @@
+// ❌ Anti-Pattern file: This code contains bad practices for educational purposes
+const request = require("supertest");
+const sinon = require("sinon");
+const nock = require("nock");
+const { initializeWebServer, stopWebServer } = require("../../example-application/api");
+const OrderRepository = require("../../example-application/data-access/order-repository");
+
+let expressApp, existingOrderId;
+
+beforeAll(async (done) => {
+  // ️️️✅ Best Practice: Place the backend under test within the same process
+  expressApp = await initializeWebServer();
+
+  const orderToAdd = {
+    userId: 1,
+    productId: 2,
+    mode: "approved",
+  };
+
+  // ❌ Anti-Pattern: Adding global records which are mutated by the tests. This will lead to high coupling and flakiness
+  //existingOrderId = (await request(expressApp).post("/order").send(orderToAdd)).body.id;
+
+  done();
+});
+
+beforeEach(() => {
+  nock("http://localhost/user/").get(`/1`).reply(200, {
+    id: 1,
+    name: "John",
+  });
+});
+
+afterEach(() => {
+  nock.cleanAll();
+  sinon.restore();
+
+  // ❌ Anti-Pattern: Cleaning here now will affect tests in other processes
+  new OrderRepository().cleanup();
+});
+
+afterAll(async (done) => {
+  // ️️️✅ Best Practice: Clean-up resources after each run
+  await stopWebServer();
+  nock.enableNetConnect();
+  done();
+});
+
+describe("/api", () => {
+  describe("POST /orders", () => {
+    test("When adding a new valid order, Then should get back 200 response", async () => {
+      //Arrange
+      const orderToAdd = {
+        userId: 1,
+        productId: 2,
+        mode: "approved",
+        externalIdentifier: "some-external-id-2", //unique value
```

Comment:
this test has no "Anti-Pattern" notation. shouln't you dismiss this line?

Reply:
It meant to create the first state that the later tests (mistakenly) rely on